Automated Smoketest: :x:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-16.ca-central-1.compute.internal
Sha: e6665949de6449ba335fe04e5f5607aadc0ba56b
Number of Error(s): 9
warn(226) A package without a profile could throw errors or warnings. package: dojo-util
warn(226) A package without a profile could throw errors or warnings. package: dojo-util
\terrors: 0
/home/centos/smoketest/PR-14280/HPCC-Platform/ecl/eclccserver/eclccserver.cpp:558:98: error: format \xe2\x80\x98%s\xe2\x80\x99 expects argument of type \xe2\x80\x98char*\xe2\x80\x99, but argument 3 has type \xe2\x80\x98StringAttr*\xe2\x80\x99 [-Werror=format=]
/home/centos/smoketest/PR-14280/HPCC-Platform/ecl/eclccserver/eclccserver.cpp:564:122: error: format \xe2\x80\x98%s\xe2\x80\x99 expects argument of type \xe2\x80\x98char*\xe2\x80\x99, but argument 3 has type \xe2\x80\x98StringAttr*\xe2\x80\x99 [-Werror=format=]
cc1plus: some warnings being treated as errors
make[2]: *** [ecl/eclccserver/CMakeFiles/eclccserver.dir/build.make:63: ecl/eclccserver/CMakeFiles/eclccserver.dir/eclccserver.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:7788: ecl/eclccserver/CMakeFiles/eclccserver.dir/all] Error 2
make: *** [Makefile:163: all] Error 2
Get ECLWatch build result
ReportTimes.
Finish at: 2020-11-05_18-17-29
time logs started
Prep time   :     9 sec (00:00:09)
Build time  :   417 sec (00:06:57)
Package time:     0 sec (00:00:00)
Install time:     0 sec (00:00:00)
Start time  :     0 sec (00:00:00)
Test time   :     0 sec (00:00:00)
Stop time   :     0 sec (00:00:00)
Summary     :   426 sec (00:07:06)
time logs end

[Build error:](https://gist.github.com/f213356e529dbb309877c9873c927ca9)
Build log [Build log](https://gist.github.com/HPCCSmoketest/f213356e529dbb309877c9873c927ca9#file-result-20-11-05-18-10-13-log)

--- a/ecl/eclccserver/eclccserver.cpp
+++ b/ecl/eclccserver/eclccserver.cpp
@@ -543,7 +543,10 @@ public:
         if (ok)
         {
             workunit->setState(WUStateCompiled);
-            const char *newClusterName = workunit->queryClusterName();   // Workunit can change the cluster name via #workunit, so reload it
+            // Workunit can change the cluster name via #workunit, so reload it
+            // Note that we need a StringAttr here not a char * as the lifetime of the value returned from queryClusterName is
+            // only until the workunit object is released.
+            StringAttr newClusterName = workunit->queryClusterName();
             if (strcmp(newClusterName, clusterName.str()) != 0)
             {
 #ifdef _CONTAINERIZED
@@ -552,13 +555,13 @@ public:
                 clusterInfo.setown(getTargetClusterInfo(newClusterName));
                 if (!clusterInfo)
                 {
-                    VStringBuffer errStr("Cluster %s by #workunit not recognized", newClusterName);
+                    VStringBuffer errStr("Cluster %s by #workunit not recognized", newClusterName.str());
                     failCompilation(errStr);
                     return;
                 }
                 if (platform != clusterInfo->getPlatform())
                 {
-                    VStringBuffer errStr("Cluster %s specified by #workunit is wrong type for this queue", newClusterName);
+                    VStringBuffer errStr("Cluster %s specified by #workunit is wrong type for this queue", newClusterName.str());
                     failCompilation(errStr);
                     return;
                 }

